File: Kth largest element in an Array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        minHeap=[]
        length=len(nums)
        for i in range(length):
            if len(minHeap)<k:
                heappush(minHeap,nums[i])
            else:
                popFirst=minHeap[0]
                if popFirst<nums[i]:
                    heappop(minHeap)
                    heappush(minHeap,nums[i])
       
        return minHeap[0]

File: KthLargestElementInAnArray.java
/**
 TC: O(n log k)
 SC: O(k)
 */

import java.util.PriorityQueue;

class Solution {

    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> minHeap = new PriorityQueue<>(k);

        for (int num : nums) {
            if (minHeap.size() < k) {
                minHeap.offer(num);
            } else if (num > minHeap.peek()) {
                minHeap.poll();
                minHeap.offer(num);
            }
        }
        return minHeap.peek();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] nums = {3, 2, 1, 5, 6, 4};
        int k = 2;
        System.out.println(solution.findKthLargest(nums, k)); // Output: 5

        nums = new int[]{3, 2, 3, 1, 2, 4, 5, 5, 6};
        k = 4;
        System.out.println(solution.findKthLargest(nums, k)); // Output: 4
    }
}
